R/pkg/inst/worker/worker.R

```diff
@@ -84,6 +84,10 @@ broadcastElap <- elapsedSecs()
 # as number of partitions to create.
 numPartitions <- SparkR:::readInt(inputCon)
 
+# If true, working for RDD
+# If false, working for DataFrame
+isDataFrame <- as.logical(SparkR:::readInt(inputCon))
```

Comment:
this might be beyond the scope of this JIRA/PR: when we have protocol changes like this, how do we make sure the peer has matching implementation, and then we are not misinterpreting the byte streams? Should there be some sort of protocol version handshake?